system/doc/reference_manual/expressions.xml

```diff
@@ -432,15 +432,17 @@ receive
 end</pre>
     <p>
       Fetches a received message present in the message queue of
-      the process. The patterns <c>Pattern</c> are sequentially matched
-      against the first message in the message queue, then the second,
-      and so on. Messages in the message queue are
+      the process. The first message in the message queue is matched
+      sequentially against the patterns from top to bottom. If no
+      match was found, the matching sequence is repeated for the second
+      message in the queue, and so on. Messages in the message queue are
       <seeguide marker="processes#message-queue-order">ordered in the
```

Comment:
Would `delivered` be better than `ordered`, here?

Reply:
I don't know. The text talks about the message queue as passive and the receive expression as active, how it matches and picks messages from the queue. So I think "delivered" is a bit confusing; from who to who?

Is it the "ordered in the order" phrasing that you don't like?

Reply:
> Is it the "ordered in the order" phrasing that you don't like?

Yes, it's odd :) but I get your point.

Reply:
Alternatives:
* "ordered as they were received"
* "kept in the order they were received"
* (longer form) "Messages are queued in the order they were received"